ndindex/tests/test_literal_eval_index.py

```diff
@@ -0,0 +1,69 @@
+import ast
+from hypothesis import given, example
+from hypothesis.strategies import one_of
+import pytest
+
+from ..literal_eval_index import literal_eval_index
+from .helpers import ellipses, ints, slices, tuples, _doesnt_raise
+
+Tuples = tuples(one_of(
+    ellipses(),
+    ints(),
+    slices(),
+)).filter(_doesnt_raise)
+
+ndindexStrs = one_of(
+    ellipses(),
+    ints(),
+    slices(),
+    Tuples,
+).map(lambda x: f'{x}')
+
+class _Dummy:
+    def __getitem__(self, x):
+        return x
+_dummy = _Dummy()
+
+@example('3')
+@example('-3')
+@example('...')
+@example('Ellipsis')
+@example('+3')
+@example('3:4')
+@example('3:-4')
+@example('3, 5, 14, 1')
+@example('3, -5, 14, -1')
+@example('3:15, 5, 14:99, 1')
+@example('3:15, -5, 14:-99, 1')
+@example(':15, -5, 14:-99:3, 1')
+@example('3:15, -5, :, [1,2,3]')
+@example('slice(None)')
+@example('slice(None, None)')
+@example('slice(None, None, None)')
+@example('slice(14)')
+@example('slice(12, 14)')
+@example('slice(12, 72, 14)')
+@example('slice(-12, -72, 14)')
+@example('3:15, -5, slice(12, -14), (1,2,3)')
+@example('..., -5, slice(12, -14), (1,2,3)')
+@example('3:15, -5, slice(12, -14), (1,2,3), Ellipsis')
```

Comment:
These are not valid indices, at least as far as ndindex is concerned. Nested tuples are not allowed.

Reply:
I did see

https://github.com/Quansight/ndindex/blob/9e5399fb20f5800474aa2bf081e8323cfeaeabf4/ndindex/tuple.py#L65

Why did you decide to be more restrictive about this syntax than `numpy`?

Reply:
The reason is that I wanted to avoid confusion from both `Tuple(1, 2)` and `Tuple((1, 2))` being valid and meaning different things. See #17. 

I also had thought that using a tuple to represent an array was deprecated, but now I can't find that.  Certainly there is ambiguity even at the NumPy level. If you expect a tuple to act like a fancy index, then you will be surprised that `a[array([1, 2])]` will give a different thing than `a[(1, 2)]`.

Reply:
Particularly this comment https://github.com/Quansight/ndindex/issues/17#issuecomment-670772519. Perhaps this should be added to the docs. I do have https://quansight.github.io/ndindex/type-confusion.html#tuple, but that is more about best practices and doesn't really discuss why things are the way they are.

Reply:
Okay, so it will be a little difficult to implement disallowing nested tuples indexes, since there are a bunch of places where we do need to parse `ast.Tuple` regardless. I think the way to do this will be to include some kind of guard in the `ast.Tuple` handling clause of `_convert` that will raise an error if it gets called twice. So far I've got:

```python
class _Guard:
    def __init__(self):
        self.val = False

    def __call__(self):
        if self.val:
            return True
        else:
            self.val = True
            return False
```

and then

```python
    _nested_tuple_guard = _Guard()
    def _convert(node):
        if isinstance(node, ast.Tuple):
            if _nested_tuple_guard():
                raise ValueError(f'tuples inside of tuple indices are not supported: {node!r}')

            return tuple(map(_convert, node.elts))
    ...
```

There's probably a cleaner way to implement the guard

Reply:
I think the simpler way is to not worry about nonsense combinations here, and let `ndindex()` handle it at the end.